### PR TITLE
MOD-797 Implement getMyUserSSO query

### PR DESCRIPTION
**Pull request by paleksandrs**
Implemented new query
```
query: getMyUserSSO: UserSSO
type UserSSO: {
  id: ID
  sub: String
}
```

### CODE REVIEW

File: Identity/Identity.swift [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import Foundation
 
 public protocol Identity {
+    var sso: SSOProvidable { get }
```
Comment (rossrossp) [inline Identity/Identity.swift:12]: Maybe something more explicit...
ssoProvider? ssoAccountDetailProvider?

File: Identity/IdentityFactory.swift [hunk 2]
```diff
@@ -11,8 +11,11 @@ import Foundation
...
     static func makeModule(
         analyticsLogger: Analytics,
-        identityPersister: IdentityPersisting
+        identityPersister: IdentityPersisting,
+        graphQLManager: GraphQLManageable
     ) -> Identity {
-        return IdentityImplementing(analyticsLogger: analyticsLogger, identityPersister: identityPersister)
+        return IdentityImplementing(analyticsLogger: analyticsLogger,
+                                    identityPersister: identityPersister,
+                                    sso: SSORepository(graphQLManager: graphQLManager))
```
Comment (rossrossp) [inline Identity/IdentityFactory.swift:19]: Should the Factory create the SSORepository or be passed the SSORepository? What pattern do we use normally?

File: Identity/Tests/IdentityImplementingTests.swift [hunk 3]
```diff
@@ -15,7 +15,9 @@ class IdentityImplementingTests: XCTestCase {
         let spy = MockAnalyticsLogger()
         let defaults = UserDefaults(suiteName: "test_identify_logged_userId_Persisted") ?? .standard
         let identityPersister = IdentityPersister(defaults: defaults)
-        let sut = IdentityFactory.makeModule(analyticsLogger: spy, identityPersister: identityPersister)
+        let sut = IdentityFactory.makeModule(analyticsLogger: spy,
+                                             identityPersister: identityPersister,
+                                             graphQLManager: MockGraphQLManager<Void>())
```
Comment (rossrossp) [inline Identity/Tests/IdentityImplementingTests.swift:20]: Should we not test that the SSOProvider is set?

File: Identity/Identity.swift (hunk 1 above)
Comment (paleksandrs) [reply Identity/Identity.swift:12]: I would recommend leaving with the suffix `able`. From [Swift Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/#protocols-describing-what-is-should-read-as-nouns)
> Protocols that describe a capability should be named using the suffixes able, ible, or ing (e.g. Equatable, ProgressReporting).

Also it seems that this naming is used inside the SDK already, e.g. `ProductProvidable`, `PaymentProvidable`. 
Should it be `ssoAccountDetailProvider`...it depends if we want to mirror Android names, I think they have this pass: 
`identity.sso.SSOFeature`, but we don't use `Feature` anywhere, so to mirror it, I used just `sso`  Do you want me to change it?

File: Identity/IdentityFactory.swift (hunk 2 above)
Comment (paleksandrs) [reply Identity/IdentityFactory.swift:19]: The factory should create `SSORepository`. The same approach is used in `SellFactory`

File: Identity/Tests/IdentityImplementingTests.swift (hunk 3 above)
Comment (paleksandrs) [reply Identity/Tests/IdentityImplementingTests.swift:20]: do you mean checking if `sut.sso` not nil? I can add that test, but `sso` property is not optional. Do you think we still need to test it?

File: Identity/Identity.swift (hunk 1 above)
Comment (rossrossp) [reply Identity/Identity.swift:12]: I meant more the variable name than the class name!
"sso" is a bit vague!

File: Identity/IdentityFactory.swift (hunk 2 above)
Comment (rossrossp) [reply Identity/IdentityFactory.swift:19]: Thanks

File: Identity/Tests/IdentityImplementingTests.swift (hunk 3 above)
Comment (rossrossp) [reply Identity/Tests/IdentityImplementingTests.swift:20]: OK, that's fine then

File: Identity/Identity.swift (hunk 1 above)
Comment (paleksandrs) [reply Identity/Identity.swift]: Oh, got it. Agree. Renamed it

**Comment by rossrossp** [comment]
Approved

**Review by rossrossp — APPROVED** [review]

**Pull request merged**